intercepting-filter/src/main/java/com/iluwatar/FilterChain.java

```diff
@@ -0,0 +1,49 @@
+ package com.iluwatar;
+ import java.util.*;
+/**
+ * Filter Chain carries multiple filters and help to execute them in defined order on target.
+ * 
+ * @author joshzambales
+ */
+public class FilterChain{
+	private ArrayList<Filter> filters = new ArrayList<Filter>();
+	private final Target target;
+
+	public FilterChain(Target target){
+		this.target = target;
+	}
+	public void addFilter(Filter filter){
+		filters.add(filter);
+	}
+
+	public String execute(String request){
+		String tempout[] = new String[filters.size()];
+
+		String tempin[] = request.split("&");
+		int i = 0;
+		try{
+			for(Filter filter:filters){
+				tempout[i] = null;
+				tempout[i++] = filter.execute(tempin);
+			}
+		}catch(Exception e){
+			return "NOT ENOUGHT INPUT";
+		}
+		
+		if(tempout[4] == null){
+			return "INVALID ORDER!";
+		}else if(tempout[3] == null){
+			return "INVALID DEPOSIT NUMBER!";
+		}else if(tempout[2] == null){
+			return "INVALID ADRDESS!";
+		}else if(tempout[1] == null){
+			return "INVALID Contact Number!";
+		}else if(tempout[0] == null){
+			return "INVALID Name!";
```

Comment:
Don't know if this will violate pattern structure or filters reusability, but I would like to keep related things together.
For example OrderFilter must know that it works with input array index-field "4", and FilterChain must know that value abscence for tempout[4] is "INVALID ORDER".
I see this magic (numbers and messages) go as constants either to Filter interface, or as public static final fields in each implementation.
```if (tempout[OrderFilter.INDEX] == null) { return OrderFilter.ERROR_MSG; }```

And one step further (although its also can have more problem with pattern consistency):
can we move INDEX and ERROR_MSG to Filter interface as methods, so we can perform polimorfic loop on list of filters instead of if-else chain?